lib/charms/kratos/v0/kratos_info.py

```diff
@@ -0,0 +1,169 @@
+        # fetch the relation info
+        try:
+            kratos_data = self.kratos_info_relation.get_kratos_info()
+        except KratosInfoRelationError as error:
+            ...
+```
+"""
+
+import logging
+from typing import Dict, Optional
+
+from ops.charm import CharmBase, RelationCreatedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+
+# The unique Charmhub library identifier, never change it
+LIBID = "40d36890fe6d40409ccee34aa9245d4a"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+RELATION_NAME = "kratos-info"
+INTERFACE_NAME = "kratos_info"
+logger = logging.getLogger(__name__)
+
+
+class KratosInfoRelationReadyEvent(EventBase):
+    """Event to notify the charm that the relation is ready."""
+
+
+class KratosInfoProviderEvents(ObjectEvents):
+    """Event descriptor for events raised by `KratosInfoProvider`."""
+
+    ready = EventSource(KratosInfoRelationReadyEvent)
+
+
+class KratosInfoProvider(Object):
+    """Provider side of the kratos-info relation."""
+
+    on = KratosInfoProviderEvents()
+
+    def __init__(self, charm: CharmBase, relation_name: str = RELATION_NAME):
+        super().__init__(charm, relation_name)
+
+        self._charm = charm
+        self._relation_name = relation_name
+
+        events = self._charm.on[relation_name]
+        self.framework.observe(
+            events.relation_created, self._on_info_provider_relation_created
+        )
+
+    def _on_info_provider_relation_created(self, event: RelationCreatedEvent) -> None:
+        self.on.ready.emit()
+
+    def send_info_relation_data(
+        self,
+        admin_endpoint: str,
+        public_endpoint: str,
+        providers_configmap_name: str,
+        schemas_configmap_name: str,
+        configmaps_namespace: str,
+    ) -> None:
+        """Updates relation with endpoints and configmaps info."""
+        if not self._charm.unit.is_leader():
+            return
+
+        relations = self.model.relations[self._relation_name]
+        info_databag = {
+            "admin_endpoint": admin_endpoint,
+            "public_endpoint": public_endpoint,
+            "login_browser_endpoint": f"{public_endpoint}/self-service/login/browser",
+            "sessions_endpoint": f"{public_endpoint}/sessions/whoami",
+            "providers_configmap_name": providers_configmap_name,
+            "schemas_configmap_name": schemas_configmap_name,
+            "configmaps_namespace": configmaps_namespace,
+        }
+
+        for relation in relations:
+            relation.data[self._charm.app].update(info_databag)
+
+
+class KratosInfoRelationError(Exception):
+    """Base class for the relation exceptions."""
+    pass
+
+
+class KratosInfoRelationMissingError(KratosInfoRelationError):
+    """Raised when the relation is missing."""
```

Comment:
suggestion: how about we remove the word `Kratos` from errors so it's less redundant?

Reply:
I would prefer to keep it to avoid confusions with other similar interfaces